[DOCS] Adds data frames section under limitations (#371)
elastic/stack-docs: docs/en/stack/data-frames/limitations.asciidoc

[[dataframe-limitations]]
== {dataframe-cap} limitations

beta[]

The following limitations and known problems apply to the 7.2 release of 
the Elastic {dataframe} feature:

[float]
[[df-wizard-limitations]]

> === {dataframe-cap} wizard limitation

Reviewer: Data frames does not (yet) support fields containing arrays - in the UI or the API.

If you try to create one, the UI will fail to show the source index table (which is not the main point of this limitation)

Reviewer 2: I was able to create them (or so I thought) while playing with the eCommerce data. Are they unsupported nonetheless?

If you have fields that contain arrays, you cannot select these fields in the source 
index table in the {kib} {dataframe} wizard. You can, however, use those fields 
when you group and aggregate your data in the wizard.

[float]
[[df-ccs-limitations]]
=== {ccs-cap} limitation

> {ccs-cap} is not supported in 7.2 for {dataframe-transforms}.

Reviewer: This is by design ... not sure if we need this documented as a limitations, as it should be documented against the `transform_id`.

- only allow lowercase alphanumeric (a-z and 0-9), hyphens or underscores
- must start and end with alphanumeric
- maximum length of 64 characters

[float]
[[df-kibana-limitations]]

> === Displaying {dataframe-transforms} in {kib} limitation

Reviewer: Change title to `{dataframe-transforms} in {kib} will only display up to 100 {dataframe-transforms}`

Reviewer: (note - this will increase to 1000 with 7.2.1)

The {kib} *Machine Learning* > *Data Frames* page lists up to 100 {dataframe-transforms}.
You can work-around this limitation by calling the 
{ref}/get-data-frame-transform.html[GET {dataframe-transforms} API] 
with the `size` parameter.

[float]
[[df-dateformat-limitations]]
=== Date histogram limitation

> When you use the {ref}/data-frame-apis.html[{dataframe-transforms} API] and a date 

Reviewer 2: ```suggestion
If you use a {ref}/search-aggregations-bucket-datehistogram-aggregation.html[date
```

> histogram is used as a group by field, the defined interval and the used time format 

Reviewer 2: ```suggestion
histogram] in the `group_by` object in the create or preview {dataframe-transform} APIs, the defined interval and time format 
```

> APIs, the defined interval and time format must have the same time fidelity, 

Reviewer 2: Not a hard requirement, but can shorten this sentence as follows:
```suggestion
APIs, the defined interval and time format must have the same time fidelity.
```

> otherwise it might cause issues in the {dataframe}.

Reviewer 2: ```suggestion
Otherwise, it might cause issues in the {dataframe}.
```

> For example, if you set the interval of the date histogram to one minute (1m), 

Reviewer 2: ```suggestion
For example, if you set the `calendar_interval` of the date histogram to one minute (`1m`), 
```

> then make sure that the time format is `yyyy-MM-dd HH:mm` instead of 

Reviewer 2: ```suggestion
then make sure that the `format` is `yyyy-MM-dd HH:mm` instead of 
```

(`1m`), then make sure that the `format` is `yyyy-MM-dd HH:mm` instead of 
`yyyy-MM-dd HH:00`.

[float]
[[df-aggresponse-limitations]]

> === Strings in aggregation responses

Reviewer: This particular issue has been fixed.. however there is a window for some similar silent failures... perhaps change to something along the lines of  ..

`Aggregation responses may be incompatible with destination index mappings`

`Data frames use composite aggregations in order to transform data. In some cases, composite aggregations may return responses which are not compatible with the mappings set for the destination index. For example "NaN", "Infinity" or possibly a numeric overflow. Where possible, a null response has been substituted. Please check elasticsearch logs if you think this may have occurred. As a workaround, using the API, you may define custom destination index mappings prior to starting the data frame transform.`

If an aggregation response contains a string (for example, due to a typo in the 
field name that results in a "NaN" response), the {dataframe} will fail to index 
the value without warning or causing an error in the {es} logs. However, it results 
in an empty destination index.

> This is a https://github.com/elastic/elasticsearch/issues/43194[known issue] that 

Reviewer 2: I feel a little uncomfortable promising that an issue will be fixed--I'm sure it will be but maybe it's better to just say something like this:
```suggestion
See https://github.com/elastic/elasticsearch/issues/43194[Issue #43194].
```
